gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/search/lucene/transformer/IndexableApiDocumentTransformer.java

```diff
@@ -176,7 +177,24 @@ private boolean accept(IndexableApi indexableApi) {
     }
 
     private void transformV4Api(Document doc, IndexableApi api) {
-        var apiDefinitionV4 = api.getApi().getApiDefinitionHttpV4();
+        var apiDefinitionV4 = api.getApi().getType() == ApiType.NATIVE
+            ? api.getApi().getApiDefinitionNativeV4()
+            : api.getApi().getApiDefinitionHttpV4();
+
+        if (api.getApi().getType() != ApiType.NATIVE) {
+            transformV4ApiHttpListeners(doc, api.getApi().getApiDefinitionHttpV4());
```

Comment:
don't we need to do the same for "native" listeners?

Reply:
the `transformV4HttpListeners` only transforms the `ListenerType.HTTP` listeners. So not necessary for Native since the only possible listener type for Native is `ListenerType.KAFKA`.